pennylane/circuit_graph.py

```diff
@@ -346,18 +364,80 @@ def layers(self):
 
         return layers
 
-    def iterate_layers(self):
+    def iterate_parametrized_layers(self):
         """Parametrized layers of the circuit.
 
         Returns:
             Iterable[LayerData]: layers with extra metadata
         """
         # iterate through each layer
-        for ops, param_inds in self.layers:
+        for ops, param_inds in self.parametrized_layers:
             pre_queue = self.ancestors_in_order(ops)
             post_queue = self.descendants_in_order(ops)
             yield LayerData(pre_queue, ops, tuple(param_inds), post_queue)
 
+    def greedy_layers(self):
+        """Greedily collected layers of the circuit. Empty slots are filled with ``None``.
```

Comment:
It might be good to have an explanation here of how this differs from `parametrized_layers`

Reply:
I actually don't exactly get what `parametrized_layers` does. Maybe you could add a more detailed docstring to that method? I'll expand the one for greedy layers.